gpii/node_modules/deviceReporter/src/DeviceGet.js

```diff
@@ -22,18 +22,45 @@
         invokers: {
             handle: {
                 funcName: "gpii.handleDeviceGet",
-                args: ["{requestProxy}", "{deviceReporter}.installedSolutionsDataSource", "{deviceReporter}.platformReporter"]
+                args: [
+                    "{requestProxy}",
+                    "{deviceReporter}.installedSolutionsDataSource",
+                    "{deviceReporter}.platformReporter",
+                    "{deviceReporter}.solutionsRegistryDataSource"
+                ]
             }
         }
     });
 
-    gpii.handleDeviceGet = function (requestProxy, installedSolutionsDataSource, platformReporter) {
-        installedSolutionsDataSource.get(null, function onSuccess(solutions) {
-            requestProxy.events.onSuccess.fire({
-                solutions: solutions,
-                OS: platformReporter.reportPlatform()
+    gpii.handleDeviceGet = function (requestProxy, installedSolutionsDataSource, platformReporter, solutionsRegistryDataSource) {
```

Comment:
This isn't the right style for this integration - rather than a "FAT API" that accepts all possible arguments, we need to position the "live" device reporter as a separate grade configured at the gpii.deviceReporter level. This will replace the "installedSolutionsDataSource" component completely, rather than adding to its configuration. What I suggest is that we take the existing grade gpii.deviceReporter and rename it as gpii.deviceReporter.static, and then all the configuration you require (including the extra solutionsRegistry) can be placed into gpii.deviceReporter.live. 
Any material which is common between gpii.deviceReporter.live and gpii.deviceReporter.static can go into gpii.deviceReporter.base

Other notes: 

i) the "kettle.dataSource.simpleCallbackWrapper" was eliminated from the framework some revisions ago and can be omitted.
ii) you shouldn't duplicate the entire solutionsRegistryDataSource in any case, but you can just refer to the existing one attached to the flowManager using the reference {solutionsRegistryDataSource}. You should factor out all the code in your "else" block below into its own global function implementing gpii.deviceReporter.live.get and then just supply the "{solutionsRegistryDataSource}" in the args to its invoker.
iii) by factoring these two implementations out, you can then remove the duplication whereby each one makes a call to a dataSource and then makes a fire to requestProxy.events.onSuccess - this can all go up into the base grade.